pkg/spire/controller.go

```diff
@@ -45,6 +47,20 @@ func init() {
 // controllerKey is a way to associate the ControllerAPIClient from inside the context.Context
 type controllerKey struct{}
 
+// OnStore stores the changed spire config into the SpireClientApi
+func OnStore(ctx context.Context, logger *zap.SugaredLogger) func(name string, value interface{}) {
+	return func(name string, value interface{}) {
+		if name == config.GetSpireConfigName() {
+			cfg, ok := value.(*spireconfig.SpireConfig)
+			if !ok {
+				logger.Error("Failed to do type insertion for extracting metrics config")
+				return
+			}
+			GetControllerAPIClient(ctx).SetConfig(*cfg)
```

Comment:
Need to close connection - `	GetControllerAPIClient(ctx).Close()` and set them to:

```
sc.entryClient = nil, sc.workloadConn = nil, sc.workloadAPI = nil,  sc.serverConn = nil
```